Genre is now a dropdown with filter and sort functionality

Evan-CTL opened the pull request:
In this branch:
- I updated the JSON file so it comes pre-sorted A-Z and trimmed some whitespace to save on file size.
- Because the genre dropdowns are a lot more HTML I decided to put it into a function. That makes it easier to comply with the `max-len` eslint rule in a way that is easier to read. It also prevents potential drift between the first and second genre dropdowns.
- `getDataPoints()` was migrated out of the component to be exported for testing.

Down the road:
- I want to add an EventListener to check when a user clicks out of the dropdown area so that it closes without having to click the button again. That decision was made based off of Marc's design, but I've noticed that it's a bit distracting.
- I also recommend that we restructure the right column to be outside of the left column entirely (currently it is in the same row as the graphs. If we do that we should also remove the sticky feature on the right column otherwise it will be impossible to see the bottom when the menus are expanded.

ndittren commented:
Things are getting a little hairy in the codebase. I think we should aim to move the functions that are tested to the utils file and export from there. That can be done later but I am thinking maybe this feature should be in its own component.

The double clicking thing, let's make an executive decision here and get that to be one click. It slows everything down having to click two times for both genres in some cases. I also agree with your recommendation.

When I click popularity I still get an A-Z sorted order. I was thinking this would be not just a shorter list of the most popular genres, but that it would list the most popular genres in descending order.

Evan-CTL commented:
The bootstrap icons are being used for the up and down arrows on the fright side of the `A-Z` and `Z-A` buttons. It's pretty minor, so we could ditch them, but I wanted to recreate Marc's design. From a user experience point of view I do think it is good practice to use pictograms.

ndittren commented:
The bootstrap icons come with bootstrap. You can remove the bootstrap-icon package and it should work the same.

Evan-CTL commented:
I removed `bootstrap-icon` and moved the genre menu into `genrePicker.tsx`. As for the ordering in Popularity, it actually is ordered, but when the count is identical the order remains alphabetical for those items. I verified this with `console.log` Alte has a count of 1001 and it remains alphabetical all the way down to Wave which is also 1001. Then the alphabeticall order repeats for 1000, and so on.

ndittren commented:
Ah okay. 

I still see the bootstrap-icons on here.

One click choice should definitely be implemented in this PR. It's too clumsy having it be two clicks.

Also, the alphabetical sorting isn't working for me. I'm not sure if it's just slow on my machine but when I click it nothing happens except the icon change.

ndittren commented on package.json:
```diff
@@ -29,6 +29,7 @@
     "@types/seedrandom": "3.0.5",
     "babel-loader": "^9.1.2",
     "bootstrap": "^5.2.3",
+    "bootstrap-icons": "^1.10.5",
```
@Evan-CTL Looks good! Just remove this package and I'll merge.

Evan-CTL replied:
Oops! Thought I removed that.
